modules/programs/ripgrep.nix

```
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.ripgrep;
+in {
+  meta.maintainers = [ maintainers.pedorich-n ];
+
+  options = {
+    programs.ripgrep = {
+      enable = mkEnableOption "Ripgrep";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.ripgrep;
+        defaultText = literalExpression "pkgs.ripgrep";
+        description =
+          "Package containing the <command>ripgrep</command> program.";
+      };
+
+      configDir = mkOption {
+        type = types.str;
+        default = "${config.xdg.configHome}/ripgrep";
+        defaultText = literalExpression "\${config.xdg.configHome}/ripgrep";
+        description = ''
+          Directory where the <filename>ripgreprc</filename> file will be stored (Inside $HOME).
+        '';
+      };
+
+      config = mkOption {
+        type = with types; listOf str;
+        default = [ ];
+        defaultText = literalExpression "[]";
+        example = [ "--max-columns-preview" "--colors=line:style:bold" ];
+        description = ''
+          List of arguments to pass to ripgrep. Each line is given to ripgrep as a single command line argument verbatim.
+
+          See <link xlink:href="https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#configuration-file"/>
+          for an example configuration.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home = {
+      packages = [ cfg.package ];
+
+      file."${cfg.configDir}/ripgreprc" = mkIf (cfg.config != [ ]) {
+        text = (builtins.concatStringsSep "\n" cfg.config) + "\n";
```

Is this subject to potential quoting hazards?

---

Hm, I haven't thought of that. Let me see how this is implemented in other modules.

---

How do you think this should be done instead? I looked through modules and couldn't find anything more "clever" than `concatStringsSep` used in similar cases.

---

I think that depends on whether or not `ripgrep` would have any quoting issues.

As an example, for shell quoting (`sh`, `bash`, `zsh`, shell scripts, etc...) one usually uses `escapeShellArgs`.

---

> Every line is a shell argument, after trimming whitespace.

So one argument per line is correct.

```suggestion
        text = lib.concatLines cfg.config;
```